packages/dds/merge-tree/src/partialLengths.ts

```diff
@@ -432,6 +461,134 @@ export class PartialSequenceLengths {
 		}
 	}
 
+	private static accumulateMoveClientOverlap(
+		partialLength: PartialSequenceLength,
+		overlapMoveClientIds: number[],
+		seglen: number,
+	) {
+		if (partialLength.overlapObliterateClients) {
+			for (const clientId of overlapMoveClientIds) {
+				const overlapClientNode = partialLength.overlapObliterateClients.get(clientId);
+				if (!overlapClientNode) {
+					partialLength.overlapObliterateClients.put(clientId, { clientId, seglen });
+				} else {
+					overlapClientNode.data.seglen += seglen;
+				}
+			}
+		} else {
+			partialLength.overlapObliterateClients = PartialSequenceLengths.getOverlapClients(
+				overlapMoveClientIds,
+				seglen,
+			);
+		}
+	}
+
+	static accumulateMoveOverlapForExisting(
+		segmentLen: number,
+		segment: ISegment,
+		firstGte: PartialSequenceLength,
+		moveClientOverlap: number[],
+	) {
+		const moveClientOverlapWithoutLocal = moveClientOverlap.filter(
+			(id) => id !== segment.clientId,
+		);
+
+		PartialSequenceLengths.accumulateMoveClientOverlap(
+			firstGte,
+			moveClientOverlapWithoutLocal,
+			segmentLen,
+		);
+
+		if (moveClientOverlap.length !== moveClientOverlapWithoutLocal.length) {
+			PartialSequenceLengths.accumulateMoveClientOverlap(
+				firstGte,
+				[segment.clientId],
+				segment.wasMovedOnInsert ? -segment.cachedLength : segmentLen,
+			);
+		}
+	}
+
+	// todo: maybe pass object for args here? would cause overhead
+	// using primitive types here makes it easy to mess up args
+	private static calculateAfterInsertion(
+		segment: ISegment,
+		segmentLen: number,
+		remoteObliteratedLen: number | undefined,
+		obliterateOverlapLen: number | undefined,
+		partials: PartialSequenceLengthsSet,
+		seq: number,
+		clientId: number,
+		removeClientOverlap: number[] | undefined,
+		moveClientOverlap: number[] | undefined,
+	) {
+		const firstGte = partials.firstGte(seq);
+
+		let partialLengthEntry: PartialSequenceLength;
+		if (firstGte?.seq === seq) {
+			partialLengthEntry = firstGte;
+			// Existing entry at this seq--this occurs for ops that insert/delete more than one segment.
+			partialLengthEntry.seglen += segmentLen;
+			if (remoteObliteratedLen) {
+				partialLengthEntry.remoteObliteratedLen ??= 0;
```

Comment:
I suppose I lean towards making it optional. Default initializing to 0 would have to be done in a few places, since the interface is used by multiple classes. I think I tried to make this change at some point while addressing this review, but found that it made other parts uglier/more complex